src/herbivoresAndCarnivores.js

```diff
@@ -2,16 +2,72 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.innerHealthVar = 100;
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
+
+  set health(totalLifeDecreased) {
+    this.innerHealthVar = !this.hidden
+      ? totalLifeDecreased
+      : this.innerHealthVar;
+
+    if (this.innerHealthVar < 1) {
+      this.innerHealthVar = 0;
+
+      Animal.alive.splice(
+        Animal.alive.findIndex((e) => e === this),
+        1,
+      );
+    }
+  }
+
+  get health() {
+    return this.innerHealthVar;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    herbivore.health -= 50;
+  }
+
+  set health(totalLifeDecreased) {}
+
+  get health() {
+    return this.innerHealthVar;
+  }
```

Comment:
Default `health` behavior could be moved to the base class `Animal`

Reply:
I create a default getter class in parente class. But we have some rules applied in prettier. I make some improvisation to assure that getter health method is inherited from parent class using 'super' keyword.